Automation/Python/Using pyenv.md

```diff
@@ -0,0 +1,186 @@
+## What is pyenv
+
+`pyenv` is a tool used to manage multiple Python versions on a single system. It helps to easily install, switch between, and manage different versions of Python.
+
+
+## Why use it
+
+* **Avoid version conflicts**: Different projects may require different versions of Python. `pyenv` allows you to manage and isolate these versions, so you don’t run into compatibility issues.
+
+* **Easier testing**: You can test your code across multiple versions of Python without installing each version system-wide.
+
+* **Better control**: You can manage Python versions independently of the system's default Python, preventing changes that could affect system tools or applications that rely on a specific version of Python.
+
+
+## Before you continue with pyenv installation
+
+Make sure [Homebrew](https://brew.sh/) is installed in the correct location. 
+If you have migrated from Mac using Intel architecture to one with ARM, you might have Homebrew still installed in the old location.
+
+Run: `which brew`
+
+Expected output: `/opt/homebrew`
+
+If your output is `/usr/local`, you should re-install Homebrew.
+
+
+## How to install pyenv
+
+1. Install dependencies
+
+    ```sh
+    ➜  ~ brew install openssl readline sqlite3 xz zlib tcl-tk@8
+    ```
+
+2. Install pyenv using [pyenv-installer](https://github.com/pyenv/pyenv-installer)
+    * This should install `pyenv` and a few useful plugins like `pyenv-virtualenv`
+
+    ```sh
+    ➜  ~ curl https://pyenv.run | zsh
+    ```
+
+3. Update PATH in `/.zprofile`. Add the following:
```

Comment:
I don't have `zprofile`, instead I updated `.zshrc` file. Perhaps we can put a note about that in case there are similar differences.

Reply:
All of those configuration files have to be created, they do not exist by default.

You can change `zshrc`, and many will use that one.
However, each of those configuration files has slightly different purpose and execution behaviour. So some might be better suited for certain actions.

Here are a few useful resources:
- https://unix.stackexchange.com/questions/71253/what-should-shouldnt-go-in-zshenv-zshrc-zlogin-zprofile-zlogout
- https://mac.install.guide/terminal/zshrc-zprofile

TLDR;
_Current recommended practice:

Use `~/.zprofile` to set the PATH and EDITOR environment variables.
Use `~/.zshrc` for aliases and a custom prompt, tweaking the appearance and behavior of the terminal._

Reply:
I will check other articles for info on configuration files and update it there if needed. I don't want to explain zsh in this one

Reply:
I don't think you need to explain it, I just suggested you can add a note that some users might not have `.zprofile` file and might need to update file where `PATH` variable is defined.

Reply:
I get what you mean.

However, zprofile it the config file that is recommended to contain PATH. And it doesn't matter if you have some paths defined in other files. It will work.

But even if you do, because of that recommendation, we should not suggest files other than zprofile. Rather, people should update their config files.

That is why I think this topic requires a bit more info, and why I would stick to just saying "Use zprofile"